dart/lib/vm_service_lib.dart

```diff
@@ -1985,8 +1985,12 @@ class AllocationProfile extends Response {
   AllocationProfile._fromJson(Map<String, dynamic> json)
       : super._fromJson(json) {
     memoryUsage = createServiceObject(json['memoryUsage']);
-    dateLastAccumulatorReset = json['dateLastAccumulatorReset'];
-    dateLastServiceGC = json['dateLastServiceGC'];
+    dateLastAccumulatorReset = json['dateLastAccumulatorReset'] != null
```

Comment:
It sounds like these fields (`dateLastAccumulatorReset`) are optional in the spec.

It looks like this code is handling the case where the field value might be sent as a string? Perhaps from older clients?

You might also see if this code works:

```
dateLastAccumulatorReset = json['dateLastAccumulatorReset'] is string
    ? int.parse(json['dateLastAccumulatorReset'])
    : json['dateLastAccumulatorReset'];
```

Reply:
Modified to use is String. Will double check this logic with old and new versions of VM service protocol.